### PR TITLE
Add 'CreatedAt' to 'ApiKeys' table and display in Api keys list

### PR DESCRIPTION
**Pull request by jonashendrickx**
Remarks:

- For existing API keys, the 'CreatedAt' value will be set to the time the application was created. Scripts for both MS SQL Server and Sqlite were both locally tested.

image

### CODE REVIEW

**Review by jrmccannon — APPROVED** [review]

File: src/Service/Migrations/Mssql/20240115082612_AlterTableApiKeyDescAddColumnCreatedAt.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    /// <inheritdoc />
+    protected override void Up(MigrationBuilder migrationBuilder)
+    {
+        migrationBuilder.AddColumn<DateTime>(
+            name: "CreatedAt",
+            table: "ApiKeys",
+            type: "datetime2",
+            nullable: false,
+            defaultValue: new DateTime(1, 1, 1, 0, 0, 0, 0, DateTimeKind.Unspecified));
```
Comment (jrmccannon) [inline src/Service/Migrations/Mssql/20240115082612_AlterTableApiKeyDescAddColumnCreatedAt.cs:18]: Make nullable? or use offseet?

File: src/AdminConsole/Pages/App/Settings/ApiKeysModel.cshtml.cs [hunk 2]
```diff
@@ -39,6 +39,7 @@ public class ApiKeysModel
 
     public record ApiKey(
         string Id,
+        DateTime CreatedAt,
```
Comment (jrmccannon) [inline src/AdminConsole/Pages/App/Settings/ApiKeysModel.cshtml.cs:42]: Use DateTimeOffset maybe?

Comment (jrmccannon) [reply src/AdminConsole/Pages/App/Settings/ApiKeysModel.cshtml.cs:42]: And maybe give it a default value a better default value or make it nullable?

File: src/Service/Models/ApiKeyDesc.cs [hunk 3]
```diff
@@ -13,6 +13,8 @@ public class ApiKeyDesc : PerTenant
     public DateTime? LastLockedAt { get; set; }
     public DateTime? LastUnlockedAt { get; set; }
 
+    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
```
Comment (jrmccannon) [inline src/Service/Models/ApiKeyDesc.cs:16]: Use DateTimeOffset?

**Review by abergs — APPROVED** [review]

**Pull request merged**